File: algorithms/0682.baseball-game/solution.rs
impl Solution {
    pub fn cal_points(ops: Vec<String>) -> i32 {
        let mut scores: Vec<i32> = vec![];

        for op in ops {
            match op.as_str() {
                "C" => {
                    scores.pop();
                }
                "D" => {
                    scores.push(scores[scores.len() - 1] * 2);
                }
                "+" => {
                    scores.push(scores[scores.len() - 2] + scores[scores.len() - 1]);
                }
                _ => {
                    scores.push(op.parse::<i32>().unwrap());
                }
            }
        }

        scores.iter().sum()
    }
}

File: algorithms/0682.baseball-game/solution.go
func calPoints(ops []string) int {
	scores := []int{}

	for _, op := range ops {
		switch op {
		case "C":
			scores = scores[:len(scores)-1]
		case "D":
			score := scores[len(scores)-1] * 2
			scores = append(scores, score)
		case "+":
			score := scores[len(scores)-2] + scores[len(scores)-1]
			scores = append(scores, score)
		default:
			score, _ := strconv.Atoi(op)
			scores = append(scores, score)
		}
	}

	sum := 0
	for _, score := range scores {
		sum += score
	}
	return sum
}
